src/shared/hostname-setup.c

```diff
@@ -71,6 +72,24 @@ int shorten_overlong(const char *s, char **ret) {
         return 1;
 }
 
+static int acquire_hostname_from_credential(char **ret) {
+        _cleanup_free_ char *cred = NULL;
+        int r;
+
+        r = read_credential_with_decryption("system.hostname", (void **) &cred, /* ret_size= */ NULL);
```

Comment:
please document this new credential briefly in the systemd.system-credential(7) man page, i.e. in man/systemd.system-credentials.xml

Reply:
(somewhere close to where system.machine_id is documented too)

Reply:
Sure, I'll write the docs, I just wanted to make sure the concept itself is sound.